### PR TITLE
ORC-910: Enforce maven-dependency-plugin

### PR DESCRIPTION
**Pull request by williamhyun**
### What changes were proposed in this pull request?
This PR aims to enforce maven-dependency-plugin. 

### Why are the changes needed?
This will prevent unintended dependency changes. 

### How was this patch tested?
Pass the CIs and check the CI log.

### CODE REVIEW

File: java/bench/pom.xml [hunk 1]
```diff
@@ -468,6 +468,15 @@
...
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <configuration>
+          <ignoredDependencies>
+            <ignoredDependency>*:*</ignoredDependency>
+          </ignoredDependencies>
+        </configuration>
+      </plugin>
```
Comment (williamhyun) [inline java/bench/pom.xml:479]: `benchmark` module is not published to the Maven Central.

File: java/pom.xml [hunk 2]
```diff
@@ -306,6 +306,22 @@
             <failOnViolation>true</failOnViolation>
           </configuration>
         </plugin>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-dependency-plugin</artifactId>
+          <version>3.1.2</version>
```
Comment (williamhyun) [inline java/pom.xml:312]: 3.2.x has a bug. So, we should use the latest 3.1.x which is 3.1.2.

File: java/pom.xml [hunk 3]
```diff
@@ -306,6 +306,22 @@
...
+          <executions>
+            <execution>
+              <phase>package</phase>
+              <goals>
+                <goal>analyze-only</goal>
+              </goals>
+            </execution>
+          </executions>
+          <configuration>
+            <failOnWarning>true</failOnWarning>
```
Comment (williamhyun) [inline java/pom.xml:322]: This will enforce this plugin in `analyze` profile.

File: java/tools/pom.xml [hunk 4]
```diff
@@ -62,7 +66,7 @@
     </dependency>
     <dependency>
       <groupId>org.apache.hadoop</groupId>
-      <artifactId>hadoop-hdfs</artifactId>
+      <artifactId>hadoop-hdfs-client</artifactId>
```
Comment (williamhyun) [inline java/tools/pom.xml:69]: After ORC-603 (https://github.com/apache/orc/pull/607), we use `hadoop-hdfs-client`.

**Comment by williamhyun** [comment]
cc: @dongjoon-hyun @pgaref

**Review by dongjoon-hyun — APPROVED** [review]
Thank you, @williamhyun . It looks nice to me.

File: java/tools/pom.xml (hunk 4 above)
Comment (dongjoon-hyun) [reply java/tools/pom.xml:69]: Thank you for catching this, @williamhyun .

**Comment by dongjoon-hyun** [comment]
I verified this by checking the GitHub Action log.
```
[INFO] --- maven-dependency-plugin:3.1.2:analyze-only (default) @ orc-shims ---
[INFO] No dependency problems found
```

**Pull request merged**